### ISSUE TITLE
Add an extension for screen reader support

### ISSUE THREAD

**Issue by taylorsolomon** [BeSite/jQuery.mmenu#415]
Any level-n menus where n > 1 are still read by screen readers. It would be nice if there were an extension that added `aria-hidden = true` and other similar accessibility-specific markup.

**Comment by grugnog** [comment]
See http://slicknav.com/ for a good example of ARIA usage in this case (as well as keyboard accessibility, which mmenu seems to be missing also).

**Comment by FrDH** [comment]
I'm not an expert on accessibility-specific markup, so correct me if I'm wrong.
- non-visible content should have the attribute `aria-hidden="true"`
- visible content should have the attribute `aria-hidden="false"`
- anchors that make non-visible content visible should have the attribute `aria-haspopup="true"`

How would keyboard accessibility work? All I see in the slicknav example, is `tabindex="0"` on all anchors.

**Comment by grugnog** [comment]
@FrDH Yes - I think those are the main ARIA properties that would be needed here. Note that the aria-hidden property should be updated using JS as the user interacts (probably obvious, but just adding for clarity).

Not blocking the tab key from entering menus is probably the [most important](http://webaim.org/techniques/keyboard/) part of keyboard accessibility (there is some discussion of this in #73), I believe the tabindex=0 usage in Slicknav is to ensure tab doesn't jump in and out of the menu unexpectedly. Other keyboard accessibility is handled in JS in SlickNav - [here is an example](https://github.com/ComputerWolf/SlickNav/blob/master/jquery.slicknav.js#L232). The [Adobe Accessible Mega Menu project](https://adobe-accessibility.github.io/Accessible-Mega-Menu/) has some more extensive [code handling arrow keys](https://github.com/adobe-accessibility/Accessible-Mega-Menu/blob/master/js/jquery-accessibleMegaMenu.js) etc that may also be a useful reference. The [ARIA menu docs](http://www.w3.org/TR/wai-aria-practices/#menu) may also be a helpful reference.

**Comment by phun-ky** [comment]
And please, if you must use anchors as onclick buttons (the chevron submenu toggle for example), please add text to it for screen readers:

    <a class="mm-subopen" href="#mm-2"><span class="sr-only">Toggle submenu</span></a>

This is a quick fix, and would help alot.

**Comment by FrDH** [comment]
I've started working on a screen reader add-on, will be available with the next version.

**Comment by MrSkippy** [comment]
Make sure to label the .mm-next links or put a screen reader readable label in it (the label should be a configuration option)
